client/src/Widgets/ActivityTable.tsx

```diff
@@ -0,0 +1,254 @@
+  Tr,
+  Th,
+  Td,
+  Tbody,
+  Box,
+  HStack,
+  Menu,
+  MenuButton,
+  MenuList,
+} from "@chakra-ui/react";
+import { GoKebabVertical } from "react-icons/go";
+import { FaFolder } from "react-icons/fa";
+import { RiDraftFill } from "react-icons/ri";
+import { MdAssignment } from "react-icons/md";
+import { BsPeopleFill } from "react-icons/bs";
+import { useFetcher, useNavigate } from "react-router-dom";
+import { AssignmentStatus } from "../Tools/_framework/Paths/ActivityEditor";
+
+export default forwardRef(function ActivityTable(
+  {
+    content,
+    suppressAvatar,
+    showOwnerName,
+    showAssignmentStatus,
+    showPublicStatus,
+  }: {
+    content: {
+      cardLink?: string;
+      id: number;
+      assignmentStatus?: AssignmentStatus;
+      isFolder?: boolean;
+      isPublic?: boolean;
+      isShared?: boolean;
+      title: string;
+      ownerName?: string;
+      menuItems?: ReactElement;
+      editableTitle?: boolean;
+      autoFocusTitle?: boolean;
+      closeTime?: string;
+      authorRow?: boolean;
+    }[];
+    suppressAvatar?: boolean;
+    showOwnerName?: boolean;
+    showAssignmentStatus?: boolean;
+    showPublicStatus?: boolean;
+  },
+  ref: React.ForwardedRef<HTMLButtonElement>,
+) {
+  const navigate = useNavigate();
+  const fetcher = useFetcher();
+
+  return (
+    <TableContainer>
+      <Table>
+        <Thead>
+          <Tr borderBottom="2px solid gray">
+            <Th></Th>
+            <Th></Th>
+            {showPublicStatus ? <Th>Visibility</Th> : <Th></Th>}
+            <Th></Th>
+            {showOwnerName || !suppressAvatar ? <Th>Owner</Th> : <Th></Th>}
+            <Th></Th>
+          </Tr>
+        </Thead>
+        <Tbody>
+          {content.map(function (activity) {
+            if (activity.isFolder) {
+              showAssignmentStatus = false;
+            }
+
+            let assignmentStatusString: string =
+              activity.assignmentStatus !== null &&
+              activity.assignmentStatus !== undefined &&
+              activity.assignmentStatus !== "Unassigned"
+                ? activity.assignmentStatus
+                : "";
+
+            if (
+              activity.assignmentStatus === "Open" &&
+              activity.closeTime !== undefined
+            ) {
+              assignmentStatusString =
+                assignmentStatusString + " until " + activity.closeTime;
+            }
+
+            function saveUpdatedTitle(newTitle: string) {
+              if (newTitle !== activity.title && activity.id !== undefined) {
+                fetcher.submit(
+                  {
+                    _action: "update title",
+                    id: activity.id,
+                    cardTitle: newTitle,
+                    isFolder: Boolean(activity.isFolder),
+                  },
+                  { method: "post" },
+                );
+              }
+            }
+
+            return (
+              <Tr
+                key={activity.id + "_row"}
```

Comment:
This key may not be unique because one could have an author and an activity with the same id. Could make it something like `{(activity.authorRow ? "author": "activity") + activity.id}`.